app/shell2/shell.cpp

```diff
@@ -91,6 +91,7 @@ Shell2App::GlobalParseMode Shell2App::testParsePhrase(Command *command,
         switch(parseMode) {
         case MODE_BEGINNING:
             parseMode = MODE_BEFORE_FLAG;
+            [[fallthrough]];
```

Comment:
Hadn't seen this syntax before, pretty cool, but is it C++17 only? https://en.cppreference.com/w/cpp/language/attributes/fallthrough
We currently use C++14, not C++17. It's easier to compile with older/embedded compilers. I would consider exceptions but this seems non-essential -- were you getting compiler warnings here?

Reply:
Oh, ok. We can do `[gnu::fallthrough]` then which is C++14.